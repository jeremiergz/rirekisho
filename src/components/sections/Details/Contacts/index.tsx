import Anchor from 'components/Anchor';
import Box from 'components/primitives/Box';
import FlexBox from 'components/primitives/FlexBox';
import Text from 'components/primitives/Text';
import Email from 'components/svg/Email';
import GitHub from 'components/svg/GitHub';
import GitLab from 'components/svg/GitLab';
import LinkedIn from 'components/svg/LinkedIn';
import Phone from 'components/svg/Phone';
import Twitter from 'components/svg/Twitter';
import useTheme from 'hooks/useTheme';
import React, { CSSProperties } from 'react';
import { ResponsiveValue } from 'styled-system';

const Icons: Record<Models.PersonalContact['type'], React.FC<React.SVGProps<SVGSVGElement>>> = {
  email: Email,
  github: GitHub,
  gitlab: GitLab,
  linkedin: LinkedIn,
  phone: Phone,
  twitter: Twitter,
};

function isHTTPLink(link: string) {
  return !!link.match(/^https?:\/\//);
}

const Contacts: React.FC<ContactsProps> = ({ items, order }) => {
  const theme = useTheme();
  return (
    <FlexBox
      flex={1}
      flexDirection={{ _: 'row', tablet: 'column' }}
      height="100%"
      justifyContent="center"
      marginY={3}
      maxHeight={{ _: 32, tablet: 'unset' }}
      minWidth={224}
      order={order}
      textAlign="center"
    >
      {items
        .sort((a, b) => a.sortOrder - b.sortOrder)
        .map((item, index) => {
          const isEven = index % 2 === 0;
          const Icon = Icons[item.type];
          const iconColor = isEven ? theme.colors.secondary : theme.colors.primary;
          return (
            <FlexBox
              alignItems="center"
              justifyContent={{ _: 'center', tablet: 'flex-end' }}
              key={item.link}
              lineHeight="12px"
              marginBottom={{ _: 0, tablet: index !== items.length - 1 && 20 }}
              marginX={{ _: 1, tablet: 0 }}
            >
              <Anchor external={isHTTPLink(item.link)} href={item.link}>
                <FlexBox alignItems="center" flexDirection={{ _: 'column-reverse', tablet: 'row' }}>
                  <Text
                    color="primary"
                    display={{ _: 'none', tablet: 'block' }}
                    fontSize={18}
                    fontWeight="bold"
                    lineHeight="32px"
                  >
                    {item.label}
                  </Text>
                  <Box marginLeft={2}>
                    <Icon fill={iconColor} height={32} width={32} />
                  </Box>
                </FlexBox>
              </Anchor>
            </FlexBox>
          );
        })}
    </FlexBox>
  );
};

Contacts.displayName = 'Contacts';

export type ContactsProps = {
  items: Models.PersonalContact[];
  order?: ResponsiveValue<CSSProperties['order']>;
};
export default Contacts;
