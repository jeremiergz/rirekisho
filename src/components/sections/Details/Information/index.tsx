import Label from '@/components/common/Label';
import clsx from 'clsx';
import React from 'react';

function Information({ items }: InformationProps): JSX.Element {
  return (
    <div
      className="flex flex-1 flex-col h-full justify-center my-6 order-2 md:order-1 text-center md:text-left"
      style={{ minWidth: 224 }}
    >
      {items.map((item, index) => (
        <div className={clsx('flex flex-col', index !== items.length - 1 && 'mb-6')} key={item.label}>
          <Label title={item.label} />
          <span className="font-bold leading-4 text-primary dark:text-primary-dark text-xl uppercase">
            {item.value}
          </span>
        </div>
      ))}
    </div>
  );
}

export type InformationProps = {
  items: Models.PersonalInformation[];
};
export default Information;
