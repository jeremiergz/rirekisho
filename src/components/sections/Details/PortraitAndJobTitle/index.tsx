import Box from 'components/primitives/Box';
import FlexBox from 'components/primitives/FlexBox';
import Text from 'components/primitives/Text';
import { graphql, useStaticQuery } from 'gatsby';
import Img from 'gatsby-image';
import React, { CSSProperties } from 'react';
import { ResponsiveValue } from 'styled-system';

const PortraitAndJobTitle: React.FC<PortraitAndJobTitleProps> = ({ jobTitle, order }) => {
  const {
    imageSharp: { fluid: portrait },
  } = useStaticQuery<GraphQL.PortraitData>(graphql`
    query PortraitData {
      imageSharp(fluid: { originalName: { eq: "portrait.png" } }) {
        fluid(maxHeight: 256, maxWidth: 256) {
          ...GatsbyImageSharpFluid_withWebp
        }
      }
    }
  `);
  return (
    <FlexBox
      alignItems="center"
      flex={1}
      flexDirection="column"
      justifyContent="space-between"
      marginTop={-64}
      order={order}
    >
      <Box
        backgroundColor="white"
        borderRadius="50%"
        height={{ _: 184, laptopL: 224 }}
        minHeight={{ _: 184, laptopL: 224 }}
        position="relative"
        width={{ _: 184, laptopL: 224 }}
      >
        <Box left={0} margin="auto" position="absolute" right={0} top={16} width={{ _: 152, laptopL: 192 }}>
          <Img fluid={portrait} />
        </Box>
      </Box>
      <Text
        color="secondary"
        fontSize={40}
        fontWeight="bold"
        margin={{ _: '16px 0 0 0', tablet: '0 -16px 64px -16px' }}
        textAlign="center"
      >
        {jobTitle}
      </Text>
    </FlexBox>
  );
};

PortraitAndJobTitle.displayName = 'PortraitAndJobTitle';

export type PortraitAndJobTitleProps = {
  jobTitle: string;
  order?: ResponsiveValue<CSSProperties['order']>;
};
export default PortraitAndJobTitle;
