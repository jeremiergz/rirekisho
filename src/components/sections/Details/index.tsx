import Layout from 'components/Layout';
import Grid from 'components/primitives/Grid';
import { graphql, useStaticQuery } from 'gatsby';
import React from 'react';
import Contacts from './Contacts';
import Information from './Information';
import PortraitAndJobTitle from './PortraitAndJobTitle';

const Details: React.FC = () => {
  const {
    details: { contacts, jobTitle, personalInformation },
  } = useStaticQuery<GraphQL.DetailsDataQuery>(graphql`
    query DetailsData {
      details: personalDetailsJson {
        contacts {
          label
          link
          sortOrder
          type
        }
        jobTitle
        personalInformation {
          label
          sortOrder
          value
        }
      }
    }
  `);
  return (
    <Grid as="section" gridColumn="span 2" justifyContent="center" variant="item">
      <Layout.Content marginBottom={4} marginTop={2} paddingX={{ _: 0, laptopS: 48 }}>
        <Information items={personalInformation as Models.PersonalInformation[]} order={{ _: 2, tablet: 1 }} />
        <PortraitAndJobTitle jobTitle={jobTitle} order={{ _: 1, tablet: 2 }} />
        <Contacts items={contacts as Models.PersonalContact[]} order={3} />
      </Layout.Content>
    </Grid>
  );
};

Details.displayName = 'Details';

export default Details;
