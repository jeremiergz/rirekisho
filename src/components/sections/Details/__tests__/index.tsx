import ThemeProvider from 'components/providers/ThemeProvider';
import * as Gatsby from 'gatsby';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import contactsMockData from '__tests__/contacts.mock.json';
import Details from '..';
import mockData from './mock.json';

describe('components/sections/Details Test Suite', () => {
  beforeAll(() => {
    const graphql = jest.spyOn(Gatsby, 'graphql');
    graphql.mockImplementation(() => '');
    const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
    useStaticQuery.mockImplementation(() => ({
      ...contactsMockData,
      ...mockData,
    }));
  });

  it('matches snapshot', () => {
    const jsx = (
      <ThemeProvider>
        <Details />
      </ThemeProvider>
    );
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
