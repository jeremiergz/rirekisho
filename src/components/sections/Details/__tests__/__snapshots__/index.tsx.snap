// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sections/Details Test Suite matches snapshot 1`] = `
<section
  className="sc-AxjAm sc-AxiKw YgjST"
>
  <div
    className="sc-AxjAm kYgcoq"
    display="flex"
    width="100%"
  >
    <div
      className="sc-AxjAm nYZHI"
      display="flex"
      height="100%"
      order={
        Object {
          "_": 2,
          "tablet": 1,
        }
      }
    >
      <div
        className="sc-AxjAm kJlEdH"
        display="flex"
      >
        <label
          className="sc-AxirZ jtalun"
          color="secondary"
          fontSize={14}
          fontWeight="bold"
        >
          nationality
        </label>
        <span
          className="sc-AxirZ iLiciv"
          color="primary"
          fontSize={22}
          fontWeight="bold"
        >
          french
        </span>
      </div>
      <div
        className="sc-AxjAm kJlEdH"
        display="flex"
      >
        <label
          className="sc-AxirZ jtalun"
          color="secondary"
          fontSize={14}
          fontWeight="bold"
        >
          birth date
        </label>
        <span
          className="sc-AxirZ iLiciv"
          color="primary"
          fontSize={22}
          fontWeight="bold"
        >
          3-27-1987
        </span>
      </div>
      <div
        className="sc-AxjAm kJlEdH"
        display="flex"
      >
        <label
          className="sc-AxirZ jtalun"
          color="secondary"
          fontSize={14}
          fontWeight="bold"
        >
          civil status
        </label>
        <span
          className="sc-AxirZ iLiciv"
          color="primary"
          fontSize={22}
          fontWeight="bold"
        >
          single
        </span>
      </div>
      <div
        className="sc-AxjAm fthWpI"
        display="flex"
      >
        <label
          className="sc-AxirZ jtalun"
          color="secondary"
          fontSize={14}
          fontWeight="bold"
        >
          driver licenses
        </label>
        <span
          className="sc-AxirZ iLiciv"
          color="primary"
          fontSize={22}
          fontWeight="bold"
        >
          car | motorcycle
        </span>
      </div>
    </div>
    <div
      className="sc-AxjAm ddSreI"
      display="flex"
      order={
        Object {
          "_": 1,
          "tablet": 2,
        }
      }
    >
      <div
        className="sc-AxjAm jsYxsb"
        height={
          Object {
            "_": 184,
            "laptopL": 224,
          }
        }
        width={
          Object {
            "_": 184,
            "laptopL": 224,
          }
        }
      >
        <div
          className="sc-AxjAm hGyeb"
          width={
            Object {
              "_": 152,
              "laptopL": 192,
            }
          }
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "maxHeight": null,
                "maxWidth": null,
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              aria-hidden={true}
              style={
                Object {
                  "paddingBottom": "100%",
                  "width": "100%",
                }
              }
            />
            <img
              alt=""
              aria-hidden={true}
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAC4jAAAuIwF4pT92AAAFQ0lEQVQ4y02Ue0xURxTGL0ktlhVEqzwLWi00VkXbYrQFg6VYoy1IFURXcdG1QR7SBBFBBEEeIlWB7YKkIEUrgg8WhH3v3ruwD1gQrdJW08ZsxcS+jDH9+96dr3NX2HSSyUwmd353znfO+RiTycSIQ6fTMg8eTPmI+6dPnklsNqtMq9WoBgcHpm/dusH39nTztwdU03a7XXXnzqSsvv6MRPzWZrP5qNUaD0Ov13tWxmQy+jAzw253SAeHbrv2SjMQH/8R1q5dhWVvR9IZgdgPYpAl24uenm5QsEur1Uln7+l0Op8ZmGn2jDGzJoXFwmLr1s3wff01fkFgAB8SstgdFhpEwulcsDDA7ev7Gr9wwXy+rOwERkdHQaNQ/A/KMCxr9pANBoNiamoKSmWzEBggERYHB5Gw8BASHh5MQoMXkRVLQklsVARZFhZE/PzeIPSKUFRUKNy/f59CtR6o0Wj0vlJqt9voHx3C1wV5bn+/OYhZHkHeXRpK1r23DIXpm9F5XI7Bmly0Zm8nsdGRIhAxq1e76asEh8MBur4K32g0SViWdVmtI7BYOOGzpE2QJawmluZSsO1nMXa1Gc+5Hrx03MYTfRcedp5Exe5PEeDvR/z9JejoaBfE0GlCXBqNRsJwHCdzOp2wcByvNxhI6pZNRFOXjafaDrx0qvHvmBp/s714ZuymwCt42NOA9iNpiFuxBMFBb5LOzk4yPDzMW61WUNlkjMViUYlACuZNZpYcyEghqko5fh9Q4i/zdTwf7sc/3A28sNzAn1wvHvY2wliTjaOpGxG9NIwoW1vIyIiVp2UmAlUicFqk0+S4rXY7yZNn4kqRFNaLVfiptxV/GK5herATfTVFqM/dh+6KfGiqcnApPw3R4YtwuuIkcdjtbrPZLAKnRSBPJ2hiiH10jNSWl6B2/zZkbo7Drg2rYGs8hqmuOpzbk4j9W+JRmJqAm2Vy9BRnImPjWlzNTif6q13EbBmG0WDgvUDWbCbDVhu51taE1rwdaDiUigb5Tky2VeLxdQXutlditK0K43S9d7kGutpcVGckob9gD2EHVMTEcjAaKZBq5wmZY1k3y1mIbqAPmnOF+K33PB5drsejHy7gcW8LfrlUi2c09BdsH37uaYT+bD5qdn2C/oojhCaU3uW8IavGx8ephqz4UjI0NEhuni3Ck/52/NhSjnvKSow1VYI9lYNH7dX4tbsZzvZTUFUdRN62OKL+TklYGiVtRbF0PEmRTUxMvAJyHFFrteRKQwmcylIMFMtQ8Hkc0j+OwZ6EWFRkbYe2LhcTHWW4kJOCXHkmsdodhOo/C5QxNDsSCnVR14DZZBKojuhoOkO09TnYlxiLrSujcC7jC5xP24a02JUoy0rGQEMBkmIiSXV5KcacToE+hOpnfFXY4qA6SsVapHBhxGp1X6duUiZLJoe2bCCKrFQ0yjPQcng3ugqkOJIcTxLWvEPWRYfh26pjbtYyIjjHxmCk7cvM+tkMVDE5OSmWj+AYHRXyDx8i61csJTvjY8jupPUkJ3kjkSZ+SDbERJPA+fPIxSKpUJC2SSg+UY67k5MecxgcUvtQt2G99kX3CrEvZztneVQUHxm22F0h30G++jKJzF8Y6JbMk/BH0xP5iYvHkBy3Bn6BwV77amxs8oK8BkutSEqT5CoqLkHcmmgkUt0kfnMxd64voiOCUXcwBTZFIb4/IXNFvRXiNdjT1bU+s/rNmCvLqFT9nkMAEmbOPFlqwvsqVnl8uuZgCl+SkcS3FaRNa2tzVH01ObKOygOeBGhaK33yj5Z6GA0N3zD/AbkhiFp3PZcoAAAAAElFTkSuQmCC"
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "top": 0,
                  "transitionDelay": "500ms",
                  "width": "100%",
                }
              }
            />
            <picture>
              <source
                sizes="(max-width: 256px) 100vw, 256px"
                srcSet="/static/341a8605d32c50e08417d89b8a0972c2/ea22a/portrait.png 64w,
/static/341a8605d32c50e08417d89b8a0972c2/2c344/portrait.png 128w,
/static/341a8605d32c50e08417d89b8a0972c2/714e6/portrait.png 256w,
/static/341a8605d32c50e08417d89b8a0972c2/6a040/portrait.png 384w,
/static/341a8605d32c50e08417d89b8a0972c2/0a8d8/portrait.png 400w"
              />
              <img
                alt=""
                loading="lazy"
                onLoad={[Function]}
                sizes="(max-width: 256px) 100vw, 256px"
                src="/static/341a8605d32c50e08417d89b8a0972c2/714e6/portrait.png"
                srcSet="/static/341a8605d32c50e08417d89b8a0972c2/ea22a/portrait.png 64w,
/static/341a8605d32c50e08417d89b8a0972c2/2c344/portrait.png 128w,
/static/341a8605d32c50e08417d89b8a0972c2/714e6/portrait.png 256w,
/static/341a8605d32c50e08417d89b8a0972c2/6a040/portrait.png 384w,
/static/341a8605d32c50e08417d89b8a0972c2/0a8d8/portrait.png 400w"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 500ms",
                    "width": "100%",
                  }
                }
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<picture><source srcset=\\"/static/341a8605d32c50e08417d89b8a0972c2/ea22a/portrait.png 64w,
/static/341a8605d32c50e08417d89b8a0972c2/2c344/portrait.png 128w,
/static/341a8605d32c50e08417d89b8a0972c2/714e6/portrait.png 256w,
/static/341a8605d32c50e08417d89b8a0972c2/6a040/portrait.png 384w,
/static/341a8605d32c50e08417d89b8a0972c2/0a8d8/portrait.png 400w\\" sizes=\\"(max-width: 256px) 100vw, 256px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 256px) 100vw, 256px\\" srcset=\\"/static/341a8605d32c50e08417d89b8a0972c2/ea22a/portrait.png 64w,
/static/341a8605d32c50e08417d89b8a0972c2/2c344/portrait.png 128w,
/static/341a8605d32c50e08417d89b8a0972c2/714e6/portrait.png 256w,
/static/341a8605d32c50e08417d89b8a0972c2/6a040/portrait.png 384w,
/static/341a8605d32c50e08417d89b8a0972c2/0a8d8/portrait.png 400w\\" src=\\"/static/341a8605d32c50e08417d89b8a0972c2/714e6/portrait.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                }
              }
            />
          </div>
        </div>
      </div>
      <span
        className="sc-AxirZ hSsBoo"
        color="secondary"
        fontSize={40}
        fontWeight="bold"
      >
        Full Stack Engineer
      </span>
    </div>
    <div
      className="sc-AxjAm aaaFU"
      display="flex"
      height="100%"
      order={3}
    >
      <div
        className="sc-AxjAm EnBDI"
        display="flex"
      >
        <a
          className="sc-AxjAm sc-AxhCb frRwYA"
          fontWeight="semi-bold"
          href="tel:+33681516442"
          rel="noopener noreferrer"
          target="_self"
        >
          <div
            className="sc-AxjAm dfJEBz"
            display="flex"
          >
            <span
              className="sc-AxirZ istddW"
              color="primary"
              display={
                Object {
                  "_": "none",
                  "tablet": "block",
                }
              }
              fontSize={18}
              fontWeight="bold"
            >
              +33 6 81 51 64 42
            </span>
            <div
              className="sc-AxjAm gkFHjp"
            >
              <svg
                className="sc-AxhUy fdtWQU"
                fill="#31859a"
                height={32}
                viewBox="0 0 512 512"
                width={32}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="256"
                  cy="256"
                  fill="white"
                  r="224"
                />
                <ellipse
                  cx="256"
                  cy="385.85175"
                  rx="19.998854"
                  ry="19.99872"
                />
                <path
                  d="M311.79301 116.47659H200.20699c-6.25498 0-11.34429 4.85205-11.34429 10.8049v210.93035c0 5.95797 5.0859 10.80491 11.34429 10.80491h111.58602c6.25498 0 11.34429-4.85206 11.34429-10.80491V127.28149c-.003-5.95797-5.08931-10.8049-11.34429-10.8049z"
                />
                <path
                  d="M255.99317 0C114.61367 0 0 114.62144 0 255.99488S114.61367 512 255.99317 512C397.37268 512 512 397.37515 512 255.99488S397.37268 0 255.99317 0zm99.59832 382.32917c0 19.55499-15.85504 35.41163-35.41186 35.41163H191.81354c-19.55511 0-35.41186-15.85493-35.41186-35.41163V128.55125c0-19.55498 15.85504-35.411623 35.41186-35.411623h128.36609c19.55511 0 35.41186 15.854933 35.41186 35.411623z"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
      <div
        className="sc-AxjAm EnBDI"
        display="flex"
      >
        <a
          className="sc-AxjAm sc-AxhCb frRwYA"
          fontWeight="semi-bold"
          href="https://linkedin.com/in/jeremiergz"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="sc-AxjAm dfJEBz"
            display="flex"
          >
            <span
              className="sc-AxirZ istddW"
              color="primary"
              display={
                Object {
                  "_": "none",
                  "tablet": "block",
                }
              }
              fontSize={18}
              fontWeight="bold"
            >
              /jeremiergz
            </span>
            <div
              className="sc-AxjAm gkFHjp"
            >
              <svg
                className="sc-AxhUy fdtWQU"
                fill="#17365c"
                height={32}
                viewBox="0 0 512 512"
                width={32}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="256"
                  cy="256"
                  fill="white"
                  r="224"
                />
                <path
                  d="M256 0C114.637 0 0 114.637 0 256s114.637 256 256 256 256-114.637 256-256S397.363 0 256 0zm-74.39 387h-62.348V199.426h62.347zm-31.173-213.188h-.406c-20.922 0-34.453-14.402-34.453-32.402 0-18.406 13.945-32.41 35.274-32.41 21.328 0 34.453 14.004 34.859 32.41 0 18-13.531 32.402-35.274 32.402zM406.423 387h-62.34V286.652c0-25.218-9.027-42.418-31.586-42.418-17.223 0-27.48 11.602-31.988 22.801-1.649 4.008-2.051 9.61-2.051 15.215V387h-62.344s.817-169.977 0-187.574h62.344v26.558c8.285-12.78 23.11-30.96 56.188-30.96 41.02 0 71.777 26.808 71.777 84.421zm0 0"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
      <div
        className="sc-AxjAm EnBDI"
        display="flex"
      >
        <a
          className="sc-AxjAm sc-AxhCb frRwYA"
          fontWeight="semi-bold"
          href="https://twitter.com/JeremieRgz"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="sc-AxjAm dfJEBz"
            display="flex"
          >
            <span
              className="sc-AxirZ istddW"
              color="primary"
              display={
                Object {
                  "_": "none",
                  "tablet": "block",
                }
              }
              fontSize={18}
              fontWeight="bold"
            >
              @JeremieRgz
            </span>
            <div
              className="sc-AxjAm gkFHjp"
            >
              <svg
                className="sc-AxhUy fdtWQU"
                fill="#31859a"
                height={32}
                viewBox="0 0 512 512"
                width={32}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="256"
                  cy="256"
                  fill="white"
                  r="224"
                />
                <path
                  d="M256 0C114.637 0 0 114.637 0 256s114.637 256 256 256 256-114.637 256-256S397.363 0 256 0zm116.887 199.602c.113 2.52.168 5.05.168 7.593 0 77.645-59.102 167.18-167.184 167.184h.004-.004c-33.184 0-64.062-9.727-90.066-26.395 4.597.543 9.277.813 14.015.813 27.532 0 52.867-9.39 72.98-25.152-25.722-.477-47.41-17.465-54.894-40.813a58.481 58.481 0 0011.043 1.063c5.363 0 10.559-.723 15.496-2.07-26.886-5.384-47.14-29.145-47.14-57.598 0-.266 0-.504.008-.75a58.354 58.354 0 0026.613 7.347c-15.777-10.527-26.148-28.523-26.148-48.91a58.597 58.597 0 017.957-29.535c28.976 35.555 72.28 58.937 121.117 61.394a58.708 58.708 0 01-1.528-13.398c0-32.437 26.317-58.754 58.766-58.754 16.902 0 32.168 7.145 42.89 18.567a117.855 117.855 0 0037.313-14.262c-4.394 13.715-13.707 25.222-25.84 32.5 11.887-1.422 23.215-4.574 33.742-9.254a119.412 119.412 0 01-29.308 30.43zm0 0"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
      <div
        className="sc-AxjAm EnBDI"
        display="flex"
      >
        <a
          className="sc-AxjAm sc-AxhCb frRwYA"
          fontWeight="semi-bold"
          href="https://gitlab.com/jeremiergz"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="sc-AxjAm dfJEBz"
            display="flex"
          >
            <span
              className="sc-AxirZ istddW"
              color="primary"
              display={
                Object {
                  "_": "none",
                  "tablet": "block",
                }
              }
              fontSize={18}
              fontWeight="bold"
            >
              gitlab/jeremiergz
            </span>
            <div
              className="sc-AxjAm gkFHjp"
            >
              <svg
                className="sc-AxhUy fdtWQU"
                fill="#17365c"
                height={32}
                viewBox="0 0 512 512"
                width={32}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="256"
                  cy="256"
                  fill="white"
                  r="224"
                />
                <path
                  d="M256 0A256 256 0 000 256a256 256 0 00256 256 256 256 0 00256-256A256 256 0 00256 0zm109.32422 94.802734c2.76229-.113738 5.58539 1.416019 6.61914 4.597657l38.81641 119.460939 19.58593 60.27344-.00195.00195a13.323818 13.323818 0 01-4.8457 14.91797L255.99414 417.20312 86.488281 294.04883c-.946726-.68776-1.754552-1.51184-2.478515-2.39453a13.309895 13.309895 0 01-2.365235-12.51953l11.382813-35.01758 8.203126-25.25 38.81445-119.455081a6.5045845 6.5045845 0 011.54883-2.490234c2.49213-2.639698 6.74639-2.677427 9.35547-.199219.2506.23947.53476.489708.75195.779297a7.6517972 7.6517972 0 011.03711 2.136719l13.09375 40.548828 25.71875 79.60938h128.88867l38.81836-119.927739c.49565-1.51754 1.39792-3.12406 2.51172-3.892579 1.05637-.73093 2.2991-1.122128 3.55469-1.173828z"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
      <div
        className="sc-AxjAm EnBDI"
        display="flex"
      >
        <a
          className="sc-AxjAm sc-AxhCb frRwYA"
          fontWeight="semi-bold"
          href="https://github.com/jeremiergz"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="sc-AxjAm dfJEBz"
            display="flex"
          >
            <span
              className="sc-AxirZ istddW"
              color="primary"
              display={
                Object {
                  "_": "none",
                  "tablet": "block",
                }
              }
              fontSize={18}
              fontWeight="bold"
            >
              github/jeremiergz
            </span>
            <div
              className="sc-AxjAm gkFHjp"
            >
              <svg
                className="sc-AxhUy fdtWQU"
                fill="#31859a"
                height={32}
                viewBox="0 0 512 512"
                width={32}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="256"
                  cy="290"
                  fill="white"
                  r="228"
                />
                <path
                  d="M343.718 509.152c-2.493.922-5.001 2.036-7.567 2.497-9.829 1.799-16.12-3.404-16.178-13.705-.13-19.132-.13-38.233.174-57.365.319-22-.174-43.718-16.512-61.869 9.713-1.783 18.353-3.032 26.804-5.024 55.986-13.185 85.067-49.397 88.778-109.497 1.899-30.592-3.421-58.657-23.253-82.754-1.435-1.694-1.507-5.544-.826-8.012 5.277-19.92 3.74-39.378-2.406-58.851-1.523-4.861-4.045-6.362-8.727-6.05-16.44.936-31.502 6.853-44.78 15.95-12.801 8.815-24.718 10.733-39.925 7.165-33.487-7.745-67.192-5.485-100.607 2.453-3.783.921-8.872-.312-12.336-2.393-15.686-9.425-31.371-18.582-49.463-22.239-13.99-2.794-15.207-2.08-18.918 11.788-4.668 17.482-5.132 34.978-.363 52.563.682 2.557-.029 6.66-1.696 8.548-37.068 42.41-23.644 122.548 1.334 153.245 19.425 23.843 45.23 34.576 73.875 40.15 5.001.952 9.974 1.68 14.975 2.572-4.29 11.446-8.147 22.253-12.51 32.882-.856 2.051-3.567 3.775-5.785 4.638-28.094 11.178-52.246 2.958-68.323-23.19-8.596-13.944-19.628-24.364-36.067-27.382-.522-.104-1.059-.386-1.537-.312-4.421 1.19-8.8 2.438-13.206 3.657 2.478 3.538 4.233 8.295 7.552 10.42 13.932 9.038 22.992 21.927 28.98 37.134 9.408 23.843 27.673 34.472 51.404 36.553 11.583.966 23.282.149 35.967.149 0 14.805.522 31.707-.218 48.534-.377 9.068-9.263 12.695-19.904 8.979-28.225-9.9-54.391-24.097-76.992-44.046C13.861 394.26-17.598 303.136 9.526 196.48 36.562 90.163 107.031 24.95 212.393 4.48c145.793-28.274 283.903 80.598 298.37 232.357 11.424 119.634-57.247 231.583-167.045 272.314zM206.275 377.759c-.087.208-.145.357-.232.565.261.045.522.09.797.134.203.015.377.386-.565-.699zm-65.96 47.286c2.276 1.442 4.45 3.27 6.915 4.058.899.298 2.566-1.798 5.509-4.028-4.088-1.59-6.495-2.988-9.032-3.345-.985-.119-2.261 2.156-3.392 3.315zm20.774 3.657c.333 1.56 3.363 3.508 5.146 3.478 1.725-.03 3.407-2.378 5.132-3.716-1.566-1.472-3.131-2.899-7.944-7.343-1.044 3.092-2.769 5.589-2.334 7.58zm25.485-6.288c-2.145.52-3.957 2.557-5.915 3.954 1.146 1.487 2.29 2.869 4.132 5.203 2.537-2.379 4.262-4.029 6.002-5.634-1.45-1.293-3.146-3.79-4.219-3.523zm-63.176-8.815c1.188 2.304 3.03 4.296 5.523 7.551 1.667-2.794 3.479-4.935 3.16-5.366-1.58-2.126-3.726-3.865-5.842-5.455-.145-.12-3.088 2.764-2.841 3.27z"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
      <div
        className="sc-AxjAm eNiSaH"
        display="flex"
      >
        <a
          className="sc-AxjAm sc-AxhCb frRwYA"
          fontWeight="semi-bold"
          href="mailto:contact@jeremierodriguez.com"
          rel="noopener noreferrer"
          target="_self"
        >
          <div
            className="sc-AxjAm dfJEBz"
            display="flex"
          >
            <span
              className="sc-AxirZ istddW"
              color="primary"
              display={
                Object {
                  "_": "none",
                  "tablet": "block",
                }
              }
              fontSize={18}
              fontWeight="bold"
            >
              contact@jeremierodriguez.com
            </span>
            <div
              className="sc-AxjAm gkFHjp"
            >
              <svg
                className="sc-AxhUy fdtWQU"
                fill="#17365c"
                height={32}
                viewBox="0 0 512 512"
                width={32}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="256"
                  cy="256"
                  fill="white"
                  r="224"
                />
                <path
                  d="M257 210c-24.814 0-45 20.186-45 45s20.186 45 45 45 45-20.186 45-45-20.186-45-45-45z"
                />
                <path
                  d="M255 0C114.39 0 0 114.39 0 255s114.39 257 255 257 257-116.39 257-257S395.61 0 255 0zm107 330c-20.273 0-38.152-10.161-49.017-25.596C299.23 319.971 279.354 330 257 330c-41.353 0-75-33.647-75-75s33.647-75 75-75c16.948 0 32.426 5.865 45 15.383V195c0-8.291 6.709-15 15-15s15 6.709 15 15v75c0 16.538 13.462 30 30 30s30-13.462 30-30c0-100.391-66.432-150-135-150-74.443 0-135 60.557-135 135s60.557 135 135 135c30 0 58.374-9.609 82.061-27.803 15.822-12.078 33.94 11.765 18.281 23.789C328.353 408.237 293.665 420 257 420c-90.981 0-165-74.019-165-165S166.019 90 257 90c82.897 0 165 61.135 165 180 0 33.091-26.909 60-60 60z"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</section>
`;
